### ISSUE TITLE
Installation error with Visual studio 2017

### ISSUE THREAD

**Issue by pommeneige** [SuperElastix/SimpleElastix#193]
HI,
I followed the instructions on https://simpleelastix.readthedocs.io in order to install the library on Windows. 
However, after several tries I keep having the same issue when building the ALL_BUILD file in Visual Studio.  

Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB6006	"cmd.exe" exited with code 1.	ITK	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	171	
Error	MSB6006	"cmd.exe" exited with code 1.	Elastix	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	171	
Error	MSB6006	"cmd.exe" exited with code 1.	SimpleITK	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets	171	

If you have any lead on what I should try I'd be very grateful. 
Thank you.

**Comment by hadasara** [comment]
HI, did you solve your problem?

**Comment by kaspermarstal** [comment]
Can you post the full log? These just means that a custom build operation failed (i.e. some CMake-generated command failed).

**Comment by qmokid** [comment]
Hi, I have encountered the same error with VS 2017.
I resolved it by setting cmake build directory to a short one.
As mentioned in [here](https://simpleelastix.readthedocs.io/GettingStarted.html#compiling-on-windows) "Visual Studio may complain during the build if the path is longer than 50 characters"

However, I encountered other linking issues.
The first one is :
transformixlib.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) const transformix::TRANSFORMIX::`vftable'" (__imp_??_7TRANSFORMIX@transformix@@6B@) referenced in function "public: __cdecl transformix::TRANSFORMIX::TRANSFORMIX(void)" (??0TRANSFORMIX@transformix@@QEAA@XZ) [D:\SimpleElastix\Elastix-build\Core\transformix.vcxproj]
7>D:\SimpleElastix\Elastix-build\bin\Release\transformix-4.8.dll : fatal error LNK1120: 1 unresolved externals [D:\SimpleElastix\Elastix-build\Core\transformix.vcxproj]

I didn't find any "vftable" in Elasix source code, I have no idea now.
Does anyone know how to resolve this? Thanks.
